vpc.tf

```diff
@@ -14,39 +14,25 @@ resource "aws_vpc" "main_vpc" {
   }
 }
 
-resource "aws_subnet" "public_subnet_1" {
-  vpc_id     = aws_vpc.main_vpc.id
-  cidr_block = element(local.public_subnet_cidr, 1)
-
-  tags = {
-    Name = "${var.environment_code}_public_subnet_1"
-  }
-}
+resource "aws_subnet" "public_subnet" {
+  count = 2
```

Comment:
instead of `count = 2` please use `count = length(public_subnet_cidr)`
the `length` function will automatically count the number of records in a list and return the proper value.
more details here: https://developer.hashicorp.com/terraform/language/meta-arguments/count

Please make this change everywhere you have count